src/WebSocketServer/WebSocketConnection.cs

```diff
@@ -13,10 +13,9 @@
 
 namespace Neo.Plugins
 {
-    internal class WebSocketConnection<TClient> : IDictionary<Guid, TClient>, IDisposable
-        where TClient : WebSocketClient, new()
```

Comment:
Need to change this back. `WebSocketClient` is a `base` class. You can extend the class by inheriting it. For example if you want to add more properties/fields or methods to it. Like a tracking number or Client's login information.

Reply:
> Need to change this back. `WebSocketClient` is a `base` class. You can extend the class by inheriting it. For example if you want to add more properties/fields or methods to it. Like a tracking number or Client's login information.

We can extend WebSocketClient, we don't need more types of WebSocketClients, sometimes simpler is better

Reply:
Yes in theory. But this also limits the API or makes more work in the future. If the code is already there or built for reusability, than the code should stay.

Reply:
> If the code is already there or built for reusability, than the code should stay.

I'm not agree, we don't need two WebSocketClients